### PR TITLE
configAPI: Implement waitForPodsReady validations

### PR DESCRIPTION
**Pull request by tenzen-y**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
I implemented the `.waitForPodsReady` validations.
Also, I consolidated the valid test cases for the `waitForPodsReady` into one case, and added fields to the defaulting tests.

#### Which issue(s) this PR fixes:

Part-of https://github.com/kubernetes-sigs/kueue/issues/2084

#### Does this PR introduce a user-facing change?

```release-note
Added validations for the "waitForPodsReady.timeout" in the Configuration.
```

### CODE REVIEW

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -152,9 +152,12 @@ apiVersion: config.kueue.x-k8s.io/v1beta1
 kind: Configuration
 waitForPodsReady:
   enable: true
+  timeout: 50s
```
Comment (mimowo) [inline pkg/config/config_test.go:155]: nit question: any reason for changes  in this file?

Comment (tenzen-y) [reply pkg/config/config_test.go:155]: I modified this test data because there are no test cases to verify if overrided parameters are propagated.

Comment (mimowo) [reply pkg/config/config_test.go:155]: Sgtm, I see this is what we do also for other fields

Comment (tenzen-y) [reply pkg/config/config_test.go:155]: That's right. Maybe we forgot to add these fields here when we extend waitForPodsReadt feature :)

**Pull request merged**